br/pkg/restore/systable_restore.go

```diff
@@ -278,6 +278,18 @@ func (rc *Client) replaceTemporaryTableToSystable(ctx context.Context, ti *model
 		return berrors.ErrUnsupportedSystemTable.GenWithStack("restoring unsupported `mysql` schema table")
 	}
 
+	// Currently, we don't support restore resource group metadata, so we need to
+	// remove the resource group related metadata in mysql.user.
+	// TODO: this function should be removed when we support backup and restore
+	// resource group.
+	if tableName == sysUserTableName {
+		sql := fmt.Sprintf("UPDATE %s SET User_attributes = JSON_REMOVE(User_attributes, '$.resource_group');",
```

Comment:
When restoring from older backup archives, this SQL may fail.

Reply:
why this sql will fail, it's noop if the user_attributes contains no resoure_group field

Reply:
```
[BR:Common:ErrUnknown]failed to execute UPDATE `__tidb_br_temporary_mysql`.`user` SET User_attributes = JSON_REMOVE(User_attributes, '$.resource_group');: [planner:1054]Unknown column 'User_attributes' in 'field list'
```
I'm not sure, perhaps earlier backups may not contain the `user_attributes` row?

Reply:
em.. User Attributes is added in #38201, I supposed it had existed longer. Maybe we shoulud check the tidb version here.